tensorly/backend/tensorflow_backend.py

```diff
@@ -18,261 +9,176 @@
+
+    @staticmethod
+    def norm(tensor, order=2, axis=None):
+        if order == 'inf':
+            order = np.inf
+        res = tf.norm(tensor, ord=order, axis=axis)
+
+        if res.shape == ():
+            return res.numpy()
+        return res
+
+    def dot(self, tensor1, tensor2):
+        return tf.tensordot(tensor1, tensor2, axes=([self.ndim(tensor1) - 1], [0]))
+
+    @staticmethod
+    def solve(lhs, rhs):
+        squeeze = []
+        if rhs.ndim == 1:
+            squeeze = [-1]
+            rhs = tf.reshape(rhs, (-1, 1))
+        res = tf.matrix_solve(lhs, rhs)
+        res = tf.squeeze(res, squeeze)
+        return res
+
+    @staticmethod
+    def truncated_svd(matrix, n_eigenvecs=None):
+        """Computes an SVD on `matrix`
+
+        Parameters
+        ----------
+        matrix : 2D-array
+        n_eigenvecs : int, optional, default is None
+            if specified, number of eigen[vectors-values] to return
+
+        Returns
+        -------
+        U : 2D-array
+            of shape (matrix.shape[0], n_eigenvecs)
+            contains the right singular vectors
+        S : 1D-array
+            of shape (n_eigenvecs, )
+            contains the singular values of `matrix`
+        V : 2D-array
+            of shape (n_eigenvecs, matrix.shape[1])
+            contains the left singular vectors
+        """
+        dim_1, dim_2 = matrix.shape
+        if dim_1 <= dim_2:
+            min_dim = dim_1
+        else:
+            min_dim = dim_2
 
-SVD_FUNS = {'numpy_svd':partial_svd, 'truncated_svd':truncated_svd}
+        if n_eigenvecs is None or n_eigenvecs > min_dim:
+            full_matrices = True
+        else:
+            full_matrices = False
+
+        S, U, V = tf.svd(matrix, full_matrices=full_matrices)
+        U, S, V = U[:, :n_eigenvecs], S[:n_eigenvecs], tf.transpose(V)[:n_eigenvecs, :]
+        return U, S, V
+
+    @property
+    def SVD_FUNS(self):
+        return {'numpy_svd': self.partial_svd,
+                'truncated_svd': self.truncated_svd}
+
+_FUN_NAMES = [
+    # source_fun, target_fun
+    (np.int32, 'int32'),
+    (np.int64, 'int64'),
+    (np.float32, 'float32'),
+    (np.float64, 'float64'),
+    (tf.ones, 'ones'),
+    (tf.zeros, 'zeros'),
+    (tf.zeros_like, 'zeros_like'),
+    (tf.eye, 'eye'),
+    (tf.reshape, 'reshape'),
+    (tf.transpose, 'transpose'),
+    (tf.where, 'where'),
+    (tf.sign, 'sign'),
+    (tf.abs, 'abs'),
+    (tf.sqrt, 'sqrt'),
+    (tf.qr, 'qr'),
+    (tf.argmin, 'argmin'),
+    (tf.argmax, 'argmax'),
+    (tf.identity, 'copy'),
+    (tf.concat, 'concatenate'),
+    (tf.reduce_min, 'min'),
+    (tf.reduce_max, 'max'),
+    (tf.reduce_mean, 'mean'),
+    (tf.reduce_sum, 'sum'),
+    (tf.reduce_prod, 'prod'),
+    (tf.reduce_all, 'all'),
+    ]
+for source_fun, target_fun_name in _FUN_NAMES:
+    TensorflowBackend.register_method(target_fun_name, source_fun)
+del _FUN_NAMES
+
+BackendManager.register_backend(TensorflowBackend())
```

Comment:
I think this needs to be fixed as well.

Reply:
Yes, this should be removed, thanks for flagging